src/CommandingOfficers/CommanderStrong.java

```diff
@@ -1,40 +1,160 @@
     COModifier strongMod = new CODamageModifier(20); // Give us a nice base power boost.
     strongMod.apply(this);
 
     COMovementModifier moveMod = new COMovementModifier();
     moveMod.addApplicableUnitType(UnitModel.UnitEnum.APC);
     moveMod.apply(this);
+
+    // Define "high-capacity" transport unit models, to be swapped in by his abilities.
+    if( null == highCapacityUnitModels )
+    {
+      UnitModel hcApc = UnitModel.clone( getUnitModel(UnitModel.UnitEnum.APC));
+      UnitModel hcLander = UnitModel.clone( getUnitModel(UnitModel.UnitEnum.LANDER));
+      UnitModel hcCopter = UnitModel.clone( getUnitModel(UnitModel.UnitEnum.T_COPTER));
+      highCapacityUnitModels = new HashMap<UnitModel.UnitEnum, UnitModel>();
+      highCapacityUnitModels.put(hcApc.type, hcApc);
+      highCapacityUnitModels.put(hcLander.type, hcLander);
+      highCapacityUnitModels.put(hcCopter.type, hcCopter);
+      for( UnitModel um : highCapacityUnitModels.values() )
+      {
+        um.holdingCapacity += 1;
+      }
+      // TODO: other non-transport types?
+    }
+
+    addCommanderAbility(new StrongArmAbility(this));
+    addCommanderAbility(new MobilizeAbility(this));
   }
 
-  public void doAbilityMinor()
+  /**
+   * StrongArm grants Strong a firepower bonus, additional mobility,
+   * and the ability to build Bazooka units from all production buildings and the HQ.
+   */
+  private static class StrongArmAbility extends CommanderAbility
   {
-    COModifier strongerMod = new CODamageModifier(20); // Another 20% with the minor ability active.
-    strongerMod.apply(this);
-    modifiers.add(strongerMod); // Add to the list so the modifier can be reverted next turn.
+    private static final String STRONGARM_NAME = "Strongarm";
+    private static final int STRONGARM_COST = 10;
+    private static final int STRONGARM_BUFF = 20;
+
+    COModifier damageMod = null;
+
+    StrongArmAbility(Commander commander)
+    {
+      super(commander, STRONGARM_NAME, STRONGARM_COST);
+
+      damageMod = new CODamageModifier(STRONGARM_BUFF);
+    }
+
+    @Override
+    protected void perform(GameMap gameMap)
+    {
+      myCommander.addCOModifier(damageMod);
+
+      // Make bazookas buildable from all production buildings.
+      UnitModel mechModel = myCommander.getUnitModel(UnitModel.UnitEnum.MECH);
+      UnitProductionModifier upm = new UnitProductionModifier(TerrainType.AIRPORT, mechModel);
+      upm.addProductionPair(TerrainType.SEAPORT, mechModel);
+      upm.addProductionPair(TerrainType.HEADQUARTERS, mechModel);
+      myCommander.addCOModifier(upm);
+
+      // Grant foot-soldiers and transports additional movement power.
+      COMovementModifier moveMod = new COMovementModifier();
+      moveMod.addApplicableUnitType(UnitModel.UnitEnum.APC);
+      moveMod.addApplicableUnitType(UnitModel.UnitEnum.INFANTRY);
+      moveMod.addApplicableUnitType(UnitModel.UnitEnum.MECH);
+      myCommander.addCOModifier(moveMod);
```

Comment:
Yo bro, this doesn't actually work still.
You told me to tell you.

Reply:
Good catch. The `COMovementModifier` was being applied to Strong's normal `UnitModel`s, not the high-capacity ones, so APCs weren't actually getting the benefit (but infantry/mech were).

`COMovementModifier` now accepts `UnitModel`s to operate on instead of unit types, which is more explicit.